subprojects/core/src/main/java/org/gradle/process/internal/JvmForkOptions.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.process.JavaForkOptions;
+
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.gradle.process.internal.util.MergeOptionsUtil.containsAll;
+import static org.gradle.process.internal.util.MergeOptionsUtil.getHeapSizeMb;
+import static org.gradle.process.internal.util.MergeOptionsUtil.normalized;
+
+/**
+ * Represents effective options for forking a Java process.
+ *
+ * Strongly relates to {@link JavaForkOptions}.
+ */
+@NonNullApi
+public class JvmForkOptions {
```

Comment:
Can this get a `toString()` for debugging?

Reply:
Good idea, I can it

Reply:
Added `toString()` to `JvmForkOptions`. 

Do you think`JvmOptions.toString()` is also needed? I didn't added it since it has many fields and it seems we didn't need it to this point.

Reply:
It might be useful, but I think it's out of scope of this PR.